tea.py

```diff
@@ -0,0 +1,674 @@
+import pybamm
+import matplotlib.pyplot as plt
+import numpy as np
+
+
+class Tea(pybamm.BaseModel):
+    """"""
+
+    def __init__(self, parameter_values=None):
+        self.parameter_values = parameter_values
+        if self.parameter_values is None:
+            print(
+                "Warning: Missing parameter values, please supply parameter values to Tea()"
+            )
+        super().__init__()
+
+    def plot_ocvs(self, parameter_values=None):
+        # plot ocv curves, average values against gravimetric capacity of cam and
+        # what if two different parameter sets are used? if, try, except?
+        # in case it is called two times after each other, the second time it will not work
+        # the solution is to call it the second time
+        parameter_values = self.parameter_values
+        try:
+            U_n = parameter_values["Negative electrode OCP [V]"]
+            U_p = parameter_values["Positive electrode OCP [V]"]
+        except KeyError:
+            print("Error: Missing required parameters")
+            return
+
+        x0, x100, y100, y0 = pybamm.lithium_ion.get_min_max_stoichiometries(
+            parameter_values
+        )
+
+        soc = pybamm.linspace(0, 1)  # cell level soc
+        x = x0 + soc * (x100 - x0)
+        y = y0 - soc * (y0 - y100)
+
+        U_n_entries = U_n(x).entries
+        U_p_entries = U_p(y).entries
+
+        plt.clf()
+        plt.figure()
+        plt.plot(soc.entries, U_p_entries - U_n_entries, label="stack")
+        plt.plot(soc.entries, U_p_entries, label="cathode")
+        plt.plot(soc.entries, U_n_entries, label="anode")
+        plt.xlabel("State of charge")
+        plt.ylabel("Open circuit potential (V)")
+        plt.legend()
+        plt.show()
+
+    def stack_energy_densities(self, input_parameter=None, print_values=False):
```

Comment:
I think there are too many options for what can be input parameters. This will need to be well documented.

Reply:
I'm not sure any of the values that are already in `parameter_values` should be allowed as inputs to these methods. To change parameters you should be able to loop over a list of parameter values, create a new `TEA` class, do the calculations and compare. There could be a separate `Comparison` class to do this.

Reply:
Inputs should be reserved for extra parameters required by the calculation that aren't in the PyBaMM sets